vignettes/queries_dataset.Rmd

```diff
@@ -0,0 +1,178 @@
+
+Variable | Scope |  Type | Example Value
+------- | ----- | ------ | ----- | ----- | -----
+**VAR_PREFIX** | The prefix used to define the grouping variables |  Character |  "SMQ01" 
+**QUERY_NAME** | The value provided to the grouping variables name|   Character | "Immune-Mediated Guillain-Barre Syndrome"
+**TERM_LEVEL** | The variable used to define the grouping. Used in conjunction with TERM_NAME |  Character | "AEDECOD"
+**TERM_NAME** | A term used to define the grouping. Used in conjunction with TERM_LEVEL |  Character | "GUILLAIN-BARRE SYNDROME"
+**TERM_ID** | A code used to define the grouping. Used in conjunction with TERM_LEVEL |  Integer | 10018767
+QUERY_ID | Id. number of the query. This could be a SMQ identifier |  Integer | 20000131
+QUERY_SCOPE | For SMQs, scope (Broad/Narrow) of the query |  Character | BROAD, NARROW, NA
+QUERY_SCOPE_NUM | For SMQs, scope (Broad/Narrow) of the query |  Integer | 1, 2, NA
+
+**Bold  variables** are required in `dataset_queries`: an error is issued if any of these variables is missing. Other variables are optional.
+
+## Required Content
+
+Each row must be unique within the dataset.
+
+As described above, the variables `VAR_PREFIX`, `QUERY_NAME`, `TERM_LEVEL`, `TERM_NAME` and `TERM_ID` are required.
+The combination of these variables will allow the creation of the grouping variable.
+
+### Input
+
+  + `VAR_PREFIX` must be a character string starting with 2 or 3 letters, followed by a 2-digits number (e.g. "CQ01").
+
+  + `QUERY_NAME` must be a non missing character string and it must be unique within `VAR_PREFIX`. 
+
+  + `TERM_LEVEL` must be a non missing character string. 
+  
+      Each value in `TERM_LEVEL` represents a variable from `dataset` used to define the grouping variables (e.g. `AEDECOD`,`AEBODSYS`, `AELLTCD`).
+    The function `derive_vars_query()` will check that each value given in `TERM_LEVEL` has a corresponding variable in the input  `dataset` and issue an error otherwise.
+
+  + `TERM_NAME` must be a character string. 
+  This **must** be populated if `TERM_ID` is missing.
+  
+  + `TERM_ID` must be an integer. 
+  This **must** be populated if `TERM_NAME` is missing.
+  
+
+
+### Output
+
+  + `VAR_PREFIX` will be used to create the grouping variable appending the suffix "NAM". This variable will now be referred to as `ABCzzNAM`: the name of the grouping variable.
+
+    E.g. `VAR_PREFIX == "SMQ01"` will create the `SMQ01NAM` variable.
+
+    
+
+
+    For each `VAR_PREFIX`, a new `ABCzzNAM` variable is created in `dataset`.
+
+
+  + [`QUERY_NAME`]{#query_name} will be used to populate the corresponding `ABCzzNAM` variable.
+
+  + `TERM_LEVEL` will be used to identify the variables from `dataset` used to perform the grouping (e.g. `AEDECOD`,`AEBODSYS`, `AELLTCD`).
+
+  + `TERM_NAME` (for character variables), `TERM_ID` (for numeric variables) will be used to identify the records meeting the criteria in `dataset` based on the variable defined in `TERM_LEVEL`.
+  
```

Comment:
Should we combine the bullet points about `QUERY_NAME` `TERM_LEVEL` and `TERM_NAME` into the **Input** section, as those describe how inputs are used to identify matching records?

Reply:
Actually, only the `TERM_XXXX` variables are used to identify the records. `QUERY_NAME` is used to populate the query variable. No change, oK?